### PR TITLE
feat: Card Component Token

### PR DESCRIPTION
**Pull request by heiyu4585**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution
https://github.com/ant-design/ant-design/issues/41884

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Card Component Token      |
| 🇨🇳 Chinese |    迁移v4 less        |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 9a1f7f5</samp>

This pull request refactors the card component style and updates the documentation for migrating less variables to component tokens. It simplifies the `ComponentToken` interface and the `genComponentStyleHook` function in `components/card/style/index.ts`. It also fixes and adds sections for `Modal` and `Card` in both English and Chinese versions of `docs/react/migrate-less-variables.md`.

### <samp>🤖 Generated by Copilot at 9a1f7f5</samp>

*  Refactor card component to use component tokens and simplify style hook ([link](https://github.com/ant-design/ant-design/pull/42061/files?diff=unified&w=0#diff-9f1c82b699dc6a75d6cc843d4e766527920ce050e82dddbb008e23a99906f8b8L3-R19),[link](https://github.com/ant-design/ant-design/pull/42061/files?diff=unified&w=0#diff-9f1c82b699dc6a75d6cc843d4e766527920ce050e82dddbb008e23a99906f8b8L105-R104),[link](https://github.com/ant-design/ant-design/pull/42061/files?diff=unified&w=0#diff-9f1c82b699dc6a75d6cc843d4e766527920ce050e82dddbb008e23a99906f8b8L125-R123),[link](https://github.com/ant-design/ant-design/pull/42061/files?diff=unified&w=0#diff-9f1c82b699dc6a75d6cc843d4e766527920ce050e82dddbb008e23a99906f8b8L148-R147),[link](https://github.com/ant-design/ant-design/pull/42061/files?diff=unified&w=0#diff-9f1c82b699dc6a75d6cc843d4e766527920ce050e82dddbb008e23a99906f8b8L368-R389))
* Add documentation for card component tokens and mapping with less variables ([link](https://github.com/ant-design/ant-design/pull/42061/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL10-R10),[link](https://github.com/ant-design/ant-design/pull/42061/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aR39-R63),[link](https://github.com/ant-design/ant-design/pull/42061/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739R39-R63))

### CODE REVIEW

File: components/card/style/index.ts [hunk 1]
```diff
@@ -122,7 +120,7 @@ const genCardActionsStyle: GenerateStyle<CardToken> = (token): CSSObject => {
       '> span': {
         position: 'relative',
         display: 'block',
-        minWidth: token.cardActionsIconSize * 2,
+        minWidth: token.fontSize * 2,
```
Comment (MadCcc) [inline components/card/style/index.ts:123]: 原有样式就不要更改了

File: docs/react/migrate-less-variables.en-US.md [hunk 2]
```diff
@@ -36,3 +36,28 @@ This document contains the correspondence between all the less variables related
...
+
+## Card
+
+<!-- prettier-ignore -->
+| Less variables | Component Token | Note |
+| --- | --- | --- |
+| `@card-head-color` | - | Deprecated for style change |
+| `@card-head-background` | - | Deprecated for style change |
+| `@card-head-font-size` | - | Deprecated for style change |
+| `@card-head-font-size-sm` | - | Deprecated for style change |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:48]: 这些应该可以加回来，可以找一下 v4 css 对应的 v5 部分

File: components/card/style/index.ts [hunk 3]
```diff
@@ -365,24 +375,39 @@ const genCardSizeStyle: GenerateStyle<CardToken> = (token): CSSObject => {
...
-});
+    cardHeadColor: token.colorText,
+    cardHeadBackground: 'transparent',
+    cardHeadFontSize: token.fontSizeLG,
+    cardHeadFontSizeSm: token.fontSize,
+    cardActionsBackground: token.colorBgContainer,
+    cardBackground: token.colorBgContainer,
+    cardRadius: token.borderRadius,
+    cardHeadExtraColor: token.colorText,
+    cardSkeletonBg: '#cfd8dc', // todo Fixed
```
Comment (heiyu4585) [inline components/card/style/index.ts:411]: 这几个fixed帮看下 改成什么

File: components/card/style/index.ts (hunk 1 above)
Comment (heiyu4585) [reply components/card/style/index.ts]: done

File: docs/react/migrate-less-variables.en-US.md (hunk 2 above)
Comment (heiyu4585) [reply docs/react/migrate-less-variables.en-US.md]: done

File: components/card/style/index.ts [hunk 4]
```diff
@@ -145,8 +155,8 @@ const genCardActionsStyle: GenerateStyle<CardToken> = (token): CSSObject => {
         },
 
         [`> ${iconCls}`]: {
-          fontSize: cardActionsIconSize,
-          lineHeight: `${cardActionsIconSize * token.lineHeight}px`,
+          fontSize,
+          lineHeight: `${fontSize * token.lineHeight}px`,
```
Comment (MadCcc) [inline components/card/style/index.ts:159]: 这里的样式不用修改

File: components/card/style/index.ts (hunk 3 above)
Comment (MadCcc) [reply components/card/style/index.ts:411]: cardSkeletionBg 应该废弃了，现在是直接用的 Skeleton 组件

File: components/card/style/index.ts [hunk 5]
```diff
@@ -1,23 +1,34 @@
...
+  cardHeadColor: string;
+  cardHeadBackground: string;
+  cardHeadFontSize: number;
+  cardHeadFontSizeSm: number;
+  cardActionsBackground: string;
+  cardBackground: string;
+  cardRadius: number;
+  cardHeadExtraColor: string;
+  cardInnerHeadPadding: number;
+  cardSkeletonBg: string;
```
Comment (MadCcc) [inline components/card/style/index.ts:27]: 新增的 token 需要用到样式里面去

File: components/card/style/index.ts (hunk 4 above)
Comment (heiyu4585) [reply components/card/style/index.ts]: done

File: components/card/style/index.ts (hunk 5 above)
Comment (heiyu4585) [reply components/card/style/index.ts]: done

File: docs/react/migrate-less-variables.zh-CN.md [hunk 6]
```diff
@@ -25,7 +25,30 @@ title: Less 变量迁移 Design Token
...
 <!-- ### Calendar 日历 -->
 
-<!-- ### Card 卡片 -->
+## Card 对话框
+
+<!-- prettier-ignore -->
+| Less variables | Component Token | Note |
+| --- | --- | --- |
+| `@card-head-color` | `headColor` | - |
+| `@card-head-background` | `headBackground` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:34]:
```suggestion
| `@card-head-background` | `headerBg` | - |
```

File: docs/react/migrate-less-variables.zh-CN.md [hunk 7]
```diff
@@ -25,7 +25,30 @@ title: Less 变量迁移 Design Token
 
 <!-- ### Calendar 日历 -->
 
-<!-- ### Card 卡片 -->
+## Card 对话框
+
+<!-- prettier-ignore -->
+| Less variables | Component Token | Note |
+| --- | --- | --- |
+| `@card-head-color` | `headColor` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:33]: 这个没有用到，如果是作用在 title 上可以用 `colorTextHeading` 代替

File: docs/react/migrate-less-variables.zh-CN.md [hunk 8]
```diff
@@ -25,7 +25,30 @@ title: Less 变量迁移 Design Token
...
+| `@card-head-background` | `headBackground` | - |
+| `@card-head-font-size` | `headFontSize` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:35]:
```suggestion
| `@card-head-font-size` | `titleFontSize` | - |
```
没有用到

File: docs/react/migrate-less-variables.zh-CN.md [hunk 9]
```diff
@@ -25,7 +25,30 @@ title: Less 变量迁移 Design Token
...
+| `@card-head-font-size-sm` | `headFontSizeSm` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:36]:
```suggestion
| `@card-head-font-size-sm` | `titleFontSizeSm` | - |
```
同上

File: docs/react/migrate-less-variables.zh-CN.md [hunk 10]
```diff
@@ -25,7 +25,30 @@ title: Less 变量迁移 Design Token
...
+| `@card-head-padding` | `headPadding` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:37]:
```suggestion
| `@card-head-padding` | `headerPadding` | - |
```
这个用 number 就好，因为会被其他地方用作 paddingTop

File: components/card/style/index.ts [hunk 11]
```diff
@@ -365,24 +374,38 @@ const genCardSizeStyle: GenerateStyle<CardToken> = (token): CSSObject => {
...
-    cardHeadPadding: token.padding,
-    cardPaddingBase: token.paddingLG,
-    cardHeadTabsMarginBottom: -token.padding - token.lineWidth,
-    cardActionsLiMargin: `${token.paddingSM}px 0`,
-    cardActionsIconSize: token.fontSize,
+export default genComponentStyleHook(
+  'Card',
+  (token) => {
+    const cardToken = mergeToken<CardToken>(token);
+    console.log('tokenboxShadowCard', token.boxShadowCard);
```
[the remark this answers is not in the document]
Comment (heiyu4585) [reply components/card/style/index.ts:382]: done

**Review by MadCcc — APPROVED** [review]

**Pull request merged**